tests/projects.rs

```diff
@@ -507,3 +510,40 @@ fn issue_801() {
 fn concurrency() {
     par_project("tests/deno-unit");
 }
+
+#[test]
+fn issue_879() {
+    let f = file_with_opt(
+        "tests/projects/issue-879/input.ts",
+        Options {
+            is_module: true,
+            config: Some(Config {
+                env: Some(Default::default()),
+                module: Some(ModuleConfig::CommonJs(Default::default())),
+                jsc: JscConfig {
+                    syntax: Some(Syntax::Typescript(TsConfig {
+                        tsx: true,
+                        decorators: true,
+                        ..Default::default()
+                    })),
+                    transform: Some(TransformConfig {
+                        legacy_decorator: true,
```

Comment:
@kdy1 I think I might have accidentally given you invalid config in the issue #879, since this one doesn't even seem to enable optimizations if I read the code correctly (`optimizer` would default to `None` as is defined at https://github.com/swc-project/swc/blob/master/src/config.rs#L572)

I tested HEAD transpiler locally, and this issue still seems to happen, although my transpiler config is not the same as it was in the issue anymore.

Reply:
I reopened it.

Reply:
Nice. Really appreciate all the work and active effort you put in